roles/hosted_cluster/tasks/main.yml

```diff
@@ -0,0 +1,127 @@
+
+- name: Create Secret resource containing ssh-key
+  kubernetes.core.k8s:
+    state: "{{ hosted_cluster_state }}"
+    definition:
+      apiVersion: v1
+      kind: Secret
+      metadata:
+        name: "sshkey-cluster-{{ hosted_cluster_name }}"
+        namespace: "{{ hosted_cluster_namespace }}"
+      stringData:
+        id_rsa.pub: "{{ hosted_cluster_template.parameters.ssh_key }}"
+
+- name: Create HostedCluster resource
+  kubernetes.core.k8s:
+    state: "{{ hosted_cluster_state }}"
+    definition:
+      apiVersion: hypershift.openshift.io/v1alpha1
+      kind: HostedCluster
+      metadata:
+        name: "{{ hosted_cluster_name }}"
+        namespace: "{{ hosted_cluster_namespace }}"
+        labels:
+          "cluster.open-cluster-management.io/clusterset": "default"
+      spec:
+        release:
+          image: "quay.io/openshift-release-dev/ocp-release:{{ hosted_cluster_template.ocp_version }}"
+        pullSecret:
+          name: "pullsecret-cluster-{{ hosted_cluster_name }}"
+        sshKey:
+          name: "sshkey-cluster-{{ hosted_cluster_name }}"
+        networking:
+          clusterNetwork:
+            - cidr: 10.132.0.0/14
+          serviceNetwork:
+            - cidr: 172.31.0.0/16
+        platform:
+          type: Agent
+          agent:
+            agentNamespace: ''
+        dns:
+          baseDomain: "{{ hosted_cluster_template.parameters.base_domain }}"
+        infraID: "{{ hosted_cluster_name }}"
+        services:
+          - service: APIServer
+            servicePublishingStrategy:
+              type: LoadBalancer
+          - service: OAuthServer
+            servicePublishingStrategy:
+              type: Route
+          - service: OIDC
+            servicePublishingStrategy:
+              type: Route
+          - service: Konnectivity
+            servicePublishingStrategy:
+              type: Route
+          - service: Ignition
+            servicePublishingStrategy:
+              type: Route
+          - service: Kubelet
+            servicePublishingStrategy:
+              type: Route
+        controllerAvailabilityPolicy: HighlyAvailable
+        infrastructureAvailabilityPolicy: HighlyAvailable
+        olmCatalogPlacement: management
+
+- name: Create NodePool resource
+  kubernetes.core.k8s:
+    state: "{{ hosted_cluster_state }}"
+    definition:
+      apiVersion: hypershift.openshift.io/v1beta1
+      kind: NodePool
+      metadata:
+        name: "nodepool-{{ hosted_cluster_name }}"
+        namespace: "{{ hosted_cluster_namespace }}"
+      spec:
+        clusterName: "{{ hosted_cluster_name }}"
+        replicas: "{{ hosted_cluster_template.parameters.number_of_nodes }}"
+        management:
+          autoRepair: false
+          upgradeType: InPlace
+        platform:
+          type: Agent
+          agent:
+            agentLabelSelector:
+              matchLabels:
+                infraenvs.agent-install.openshift.io: "{{ hosted_cluster_template.infraenv }}"
```

Comment:
Flagging this for discussion, because I think it's odd that we have ended up with `infraenv` as a template parameter. I think -- at least for POC1, and possibly going forward -- that we will also want to match agents on cluster name, and some other component will be responsible for assigning agents to a cluster (e.g., we would do that by hand for POC1).

Reply:
I agree that something more functional would be great.

About the var file used as a template, I was thinking about the  `parameters` section to be merged with the parameters provided by the user in the cluster order, so everything outside this section would be hard-coded by the template.